jlm/llvm/opt/InvariantValueRedirection.cpp

```diff
@@ -84,6 +85,21 @@ InvariantValueRedirection::RedirectInRootRegion(rvsdg::Graph & rvsdg)
       // Nothing needs to be done.
       // Delta nodes are irrelevant for invariant value redirection.
     }
+    else if (auto simple = dynamic_cast<rvsdg::SimpleNode *>(node))
+    {
+      if (dynamic_cast<const FunctionToPointerOperation *>(&simple->GetOperation()))
+      {
+        // Ignore
+      }
+      else if (dynamic_cast<const PointerToFunctionOperation *>(&simple->GetOperation()))
+      {
+        // Ignore
+      }
+      else
+      {
+        JLM_UNREACHABLE("Unhandled node type.");
+      }
+    }
```

Comment:
```
else if (is<FunctionToPointerOperation>(...) || is<PointerToFunctionOperation>(...))
{
  // Ignore
}
```

There is no need to first cast it to a simple node and then again to the concrete nodes. Also, for readability, we have the is function: https://github.com/phate/jlm/blob/6db940a6220bd6be1ba2203edceb6269114ea4b6/jlm/rvsdg/operation.hpp#L53C1-L60C2

Reply:
I still need to apply it to node->GetOperation(), though. I removed the casts to SimpleNode as I can call GetOperation also just on node.